src/lib/jobs/potoken.ts

```diff
@@ -34,46 +35,90 @@ export const poTokenGenerate = async (
+    if (!Reflect.has(globalThis, "navigator")) {
+        Object.defineProperty(globalThis, "navigator", {
+            value: dom.window.navigator,
+        });
+    }
 
-    if (!bgChallenge) {
+    const challengeResponse = await innertubeClient.getAttestationChallenge(
+        "ENGAGEMENT_TYPE_UNBOUND",
+    );
+    if (!challengeResponse.bg_challenge) {
         throw new Error("Could not get challenge");
     }
 
-    const interpreterJavascript = bgChallenge.interpreterJavascript
-        .privateDoNotAccessOrElseSafeScriptWrappedValue;
-
+    const interpreterUrl = challengeResponse.bg_challenge.interpreter_url
+        .private_do_not_access_or_else_trusted_resource_url_wrapped_value;
+    const bgScriptResponse = await getFetchClient(konfigStore)(
+        `http:${interpreterUrl}`,
+    );
+    const interpreterJavascript = await bgScriptResponse.text();
     if (interpreterJavascript) {
         new Function(interpreterJavascript)();
     } else throw new Error("Could not load VM");
 
-    const poTokenResult = await BG.PoToken.generate({
-        program: bgChallenge.program,
-        globalName: bgChallenge.globalName,
-        bgConfig,
+    const botguard = await BG.BotGuardClient.create({
+        program: challengeResponse.bg_challenge.program,
+        globalName: challengeResponse.bg_challenge.global_name,
+        globalObj: globalThis,
+    });
+
+    const webPoSignalOutput: WebPoSignalOutput = [];
+    const botguardResponse = await botguard.snapshot({ webPoSignalOutput });
+    const requestKey = "O43z0dpjhgX20SCx4KAo";
+
+    const integrityTokenResponse = await fetch(buildURL("GenerateIT", true), {
+        method: "POST",
+        headers: {
+            "content-type": "application/json+protobuf",
+            "x-goog-api-key": GOOG_API_KEY,
+            "x-user-agent": "grpc-web-javascript/0.1",
+            "user-agent": USER_AGENT,
+        },
+        body: JSON.stringify([requestKey, botguardResponse]),
     });
 
-    await BG.PoToken.generatePlaceholder(visitorData);
+    const response = await integrityTokenResponse.json() as unknown[];
+
+    if (typeof response[0] !== "string") {
+        throw new Error("Could not get integrity token");
+    }
+
+    const integrityTokenBasedMinter = await BG.WebPoMinter.create({
+        integrityToken: response[0],
+    }, webPoSignalOutput);
+
+    const sessionPoToken = await integrityTokenBasedMinter.mintAsWebsafeString(
+        visitorData,
+    );
 
-    return (await Innertube.create({
-        po_token: poTokenResult.poToken,
+    const instantiatedInnertubeClient = await Innertube.create({
+        po_token: sessionPoToken,
         visitor_data: visitorData,
         fetch: getFetchClient(konfigStore),
         cache: innertubeClientCache,
         generate_session_locally: true,
-    }));
+    });
+
+    return {
+        innertubeClient: instantiatedInnertubeClient,
+        tokenMinter: integrityTokenBasedMinter,
```

Comment:
This is the important bit to get this working with invidious-companion - I'm passing the `tokenMinter` back and it lives long-term in the Hono context, being passed around to whatever needs it. This gets swapped out when the session token regens.